controllers/openstackcluster_controller.go

```diff
@@ -150,7 +150,24 @@ func (r *OpenStackClusterReconciler) reconcileDelete(ctx context.Context, log lo
 		}
 	}
 
-	// TODO(sbueringer) Delete network/subnet/router/... if created by CAPO
+	if openStackCluster.Status.Network.Router != nil {
+		log.Info("Deleting router", "name", openStackCluster.Status.Network.Router.Name)
+		err := networkingService.DeleteRouter(openStackCluster.Status.Network)
+		if err != nil {
+			return ctrl.Result{}, errors.Errorf("failed to delete router: %v", err)
+		}
+		log.Info("OpenStack router deleted successfully")
+	}
+
+	if openStackCluster.Status.Network != nil {
```

Comment:
I don't remember clearly, do we have a chance that the network/router is not created by CPAO? 
others LGTM

Reply:
I think it's okay as the cluster network/subnet are afaik always created by us

Reply:
ok, I didn't remember clearly, will try this later on and submit PR if need :)